File: goals-progress/src/app/models/Goal.ts
import dayjs from "dayjs";
import { Category } from "./Category";
import { Progress } from "./Progress";
import { GoalStatus } from "./enums/GoalStatus";
import { GoalType } from "./enums/GoalType";

export interface Goal {
    id: number,
    name: string,
    description: string,
    currentValue: number,
    targetValue: number | null,
    customUnit: boolean,
    unit: string,
    deadline: dayjs.Dayjs | Date;
    status: GoalStatus,
    type: GoalType,
    modificationDate?: dayjs.Dayjs | Date,
    completedDate?: dayjs.Dayjs | Date,
    progresses?: Progress[],
    categories?: Category[]
}

File: Domain/Goal.cs
using Domain.Enums;

namespace Domain
{
    public class Goal
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Decimal CurrentValue { get; set; }
        public Decimal TargetValue { get; set; }
        public bool CustomUnit { get; set; }
        public string Unit { get; set; }
        public DateTime Deadline { get; set; }
        public GoalStatus Status { get; set; }
        public GoalType Type { get; set; }
        public ICollection<Progress> Progresses { get; set; }
    }
}